### PR TITLE
integration tests for methods which manipulate with objects of Group class

### PR DESCRIPTION
**Pull request by cdc895642**
integration tests for methods which manipulate with objects of Group class (TeacherGroupsManipulationServiceImpl and GroupServiceImpl classes)

### CODE REVIEW

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+public class GroupServiceImplTest {
+
+  private final String TEACHER_LOCATION_WITH_GROUPS = "OlegShvets";
+  private final String TEACHER_WITH_GROUPS = "OlegShvets";
+  private final String COORDINATOR = "DmytroPetin";
+  private final String COORDINATOR_OTHER_LOCATION = "LukasLukichich";
+  private final String TEACHER_WITHOUT_GROUPS = "NogroupsUser";
+  private final String TEACHER_LOCATION_WITHOUT_GROUPS = "NoOlegShvets";
+  @Spy
+  private GroupResourceAssembler groupResourceAssembler;
```
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: I can see that you prefer using spies over mocks. Do you really need spies here?

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+    //Act
+    groupService.deleteGroup(DELETED_GROUP_ID, COORDINATOR);
+    Group deleted = groupRepository.findOne(DELETED_GROUP_ID);
+    //Assert
+    assertNull(deleted);
+  }
+
+  @WithUserDetails(COORDINATOR)
+  @TestGroup
+  @Test
```
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: This combination of @Test and @TestGroup annotations looks pretty weird. May we include @Test annotation into TestGroup? I'd also give it some more clear naming. May be @ClearTest to highlight the nature of what is hidden behind that annotation. What do you think?

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 1 above)
Comment (cdc895642) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: i need groupResourceAssembler methods except methods which use field - repositoryEntityLinks

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 2 above)
Comment (cdc895642) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: sometimes i mark test annotation by exception class. so i deside do not include test into the grouptest annotation. Am I wrong? If you think that @cleartest is a more appropriate name - I chage it

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITHOUT_GROUPS)
+  @Test
+  public void getGroupResourceById_existGroupId_ReturnGroup() {
+    //Arrange
+    final int GROUP_ID = 1;
+    final String EXPECTED_GROUP_NAME = "DP-115";
+    //Act
+    Group group = groupService.getGroupById(GROUP_ID);
```
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: Should it be ```getGroupResourceById```?

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 2 above)
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: That's just an idea. You and the team can decide what is more appropriate :)

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 3 above)
Comment (cdc895642) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: change it. Thanks

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+  @Test(expected = AccessDeniedException.class)
+  public void deleteGroup_CoordinatorOtherLocation_ExceptionThrown() {
+    //Act
+    groupService.deleteGroup(1, COORDINATOR_OTHER_LOCATION);
+  }
+
+  @WithUserDetails(COORDINATOR)
+  @TestGroup
+  @Test
+  public void getGroupsByFilter_twoLocationExist_ReturnGroups() {
```
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: Please fix minor typo in the name from ```twoLocationExist``` to ```twoLocationsExist```

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+    final int RESULT_SIZE = (int) groups.spliterator().getExactSizeIfKnown();
+
+    //Assert
+    assertEquals(EXPECTED_SIZE, RESULT_SIZE);
+  }
+
+  @WithUserDetails(COORDINATOR)
+  @TestGroup
+  @Test
+  public void getGroupsByFilter_oneLocationExist_ReturnGroups() {
```
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: Please fix minor typo in the name from ```oneLocationExist``` to ```oneLocationsExists``` ;)

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+    newGroup.setLocation(location);
+
+    //Act
+    groupService.addGroup(newGroup, COORDINATOR_OTHER_LOCATION);
+  }
+
+  @WithUserDetails(COORDINATOR)
+  @TestGroup
+  @Test(expected = ValidationException.class)
+  public void addGroup_existName_ExceptionThrown() {
```
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: I'd suggest changing ```existName``` to ```nameAlreadyExists```

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 4 above)
Comment (cdc895642) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: fixed, thanks

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 5 above)
Comment (cdc895642) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: fixed, thanks

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+    Set<GroupResource> groups = groupService
+        .getGroupResourcesFromUserLocation(TEACHER_LOCATION_WITH_GROUPS);
+    int result = (int) groups.stream()
+        .map(GroupResource::getName)
+        .filter(name -> expected.contains(name))
+        .count();
+
+    //Assert
+    assertEquals(EXPECTED_SIZE, result);
+  }
```
Comment (svoronin) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: Overall please align all tests names to follow some single naming rule. Like ```camelCasePrepositon_camelCaseAddition_camelCaseExpectedResult```. As of now you have some variance in that:
* getGroupResourcesFromUserLocation_TeacherLocationWithGroups_ReturnGroups
* getGroupsByFilter_threeLocationOneNotExist_ReturnGroups
* getAllGroupResources_teacher_returnAllGroupResources

Also please make consistent styling within Arrange - Act - Assert blocks. I like additional blank lines between those blocks that you have in the end of the file. But they are missing in the beginning of the file

**Comment by svoronin** [comment]
In general I'd also expect tests that interact with group controller here. Not just a service test. E.g. we query group with invalid credentials and check that our endpoint returns correct error code. Or we query group with proper user and expect correct group response.

Since there is no time for the major change in scope of this PR, I'd like to request that to be done in a separate pull-request.

**Comment by cdc895642** [comment]
I started write tests for GroupController but I did not have time to do all these tests, so I deside include it in next sprint

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 6 above)
Comment (cdc895642) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: fixed

**Comment by svoronin** [comment]
Ok, That makes perfect sense then :) Please fix minor issues and this PR would be good to be merged

File: src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java (hunk 7 above)
Comment (cdc895642) [inline src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java]: fixed

**Pull request merged**